File: desktop-app-vue/src/renderer/utils/messageTypes.js
/**
 * ChatPanel 消息类型定义
 * 用于在对话列表中展示不同类型的交互
 */

// 消息类型枚举
export const MessageType = {
  USER: 'user',                          // 用户消息
  ASSISTANT: 'assistant',                // AI助手回复
  SYSTEM: 'system',                      // 系统消息（提示、通知）
  INTENT_RECOGNITION: 'intent-recognition', // 意图识别结果
  TASK_ANALYSIS: 'task-analysis',        // 任务分析中
  INTERVIEW: 'interview',                // 采访问题
  TASK_PLAN: 'task-plan',                // 任务计划
  PROGRESS: 'progress',                  // 执行进度
  ERROR: 'error',                        // 错误消息
};

// 消息角色
export const MessageRole = {
  USER: 'user',
  ASSISTANT: 'assistant',
  SYSTEM: 'system',
};

/**
 * 创建用户消息
 */
export function createUserMessage(content, conversationId = null) {
  return {
    id: `msg_${Date.now()}_user`,
    conversation_id: conversationId,
    role: MessageRole.USER,
    type: MessageType.USER,
    content,
    timestamp: Date.now(),
  };
}

/**
 * 创建助手消息
 */
export function createAssistantMessage(content, conversationId = null, metadata = {}) {
  return {
    id: `msg_${Date.now()}_assistant`,
    conversation_id: conversationId,
    role: MessageRole.ASSISTANT,
    type: MessageType.ASSISTANT,
    content,
    timestamp: Date.now(),
    metadata,
  };
}

/**
 * 创建系统消息
 */
export function createSystemMessage(content, metadata = {}) {
  return {
    id: `msg_${Date.now()}_system`,
    role: MessageRole.SYSTEM,
    type: MessageType.SYSTEM,
    content,
    timestamp: Date.now(),
    metadata,
  };
}

/**
 * 创建意图识别消息
 */
export function createIntentRecognitionMessage(intentResult) {
  return {
    id: `msg_${Date.now()}_intent`,
    role: MessageRole.SYSTEM,
    type: MessageType.INTENT_RECOGNITION,
    content: '正在分析您的需求...',
    timestamp: Date.now(),
    metadata: {
      intentResult,
    },
  };
}

/**
 * 创建任务分析消息
 */
export function createTaskAnalysisMessage(status = 'analyzing') {
  return {
    id: `msg_${Date.now()}_analysis`,
    role: MessageRole.SYSTEM,
    type: MessageType.TASK_ANALYSIS,
    content: '正在分析任务复杂度...',
    timestamp: Date.now(),
    metadata: {
      status, // analyzing, completed
    },
  };
}

/**
 * 创建采访问题消息
 */
export function createInterviewMessage(questions, currentIndex = 0) {
  return {
    id: `msg_${Date.now()}_interview`,
    role: MessageRole.ASSISTANT,
    type: MessageType.INTERVIEW,
    content: '为了更好地完成任务，我需要了解一些信息：',
    timestamp: Date.now(),
    metadata: {
      questions,        // 问题列表
      currentIndex,     // 当前问题索引
      answers: {},      // 用户答案
    },
  };
}

/**
 * 创建任务计划消息
 */
export function createTaskPlanMessage(plan) {
  return {
    id: `msg_${Date.now()}_plan`,
    role: MessageRole.ASSISTANT,
    type: MessageType.TASK_PLAN,
    content: '我已经为您制定了执行计划：',
    timestamp: Date.now(),
    metadata: {
      plan,            // 任务计划对象
      status: 'pending', // pending, confirmed, executing, completed
    },
  };
}

/**
 * 创建进度消息
 */
export function createProgressMessage(taskName, progress = 0) {
  return {
    id: `msg_${Date.now()}_progress`,
    role: MessageRole.SYSTEM,
    type: MessageType.PROGRESS,
    content: taskName,
    timestamp: Date.now(),
    metadata: {
      progress, // 0-100
    },
  };
}

/**
 * 创建错误消息
 */
export function createErrorMessage(error) {
  return {
    id: `msg_${Date.now()}_error`,
    role: MessageRole.SYSTEM,
    type: MessageType.ERROR,
    content: typeof error === 'string' ? error : error.message,
    timestamp: Date.now(),
    metadata: {
      error,
    },
  };
}

File: android-app/app/src/main/java/com/chainlesschain/android/remote/model/MessageTypes.kt
package com.chainlesschain.android.remote.model

import androidx.compose.runtime.Immutable
import com.chainlesschain.android.remote.commands.TokenUsage
import kotlinx.serialization.Serializable
import java.util.UUID

/**
 * Context mode for AI chat
 * Aligned with PC desktop-app-vue ChatPanel.vue
 */
enum class ContextMode {
    /** Project context - AI has access to entire project */
    PROJECT,
    /** File context - AI focuses on specific file(s) */
    FILE,
    /** Global context - General AI conversation */
    GLOBAL
}

/**
 * File reference in chat messages
 */
@Serializable
data class FileReference(
    val id: String = UUID.randomUUID().toString(),
    val path: String,
    val name: String,
    val extension: String? = null,
    val size: Long? = null,
    val isDirectory: Boolean = false
) {
    companion object {
        fun fromPath(path: String): FileReference {
            val name = path.substringAfterLast("/").substringAfterLast("\\")
            val extension = if (name.contains(".")) name.substringAfterLast(".") else null
            return FileReference(
                path = path,
                name = name,
                extension = extension
            )
        }
    }
}

/**
 * Task plan data structure
 */
@Immutable
@Serializable
data class TaskPlan(
    val id: String = UUID.randomUUID().toString(),
    val title: String,
    val summary: String,
    val tasks: List<TaskItem>,
    val estimatedDuration: String? = null,
    val outputs: List<String> = emptyList(),
    val createdAt: Long = System.currentTimeMillis()
)

/**
 * Individual task item in a plan
 */
@Immutable
@Serializable
data class TaskItem(
    val id: Int,
    val name: String,
    val description: String,
    val action: String,
    val output: String,
    val status: TaskStatus = TaskStatus.PENDING,
    val progress: Int = 0,
    val error: String? = null
)

/**
 * Task execution status
 */
enum class TaskStatus {
    PENDING,
    IN_PROGRESS,
    COMPLETED,
    FAILED,
    SKIPPED
}

/**
 * Planning session state
 */
enum class PlanningState {
    IDLE,
    ANALYZING,
    INTERVIEWING,
    PLANNING,
    CONFIRMING,
    EXECUTING,
    COMPLETED,
    CANCELLED
}

/**
 * Interview question for task clarification
 */
@Immutable
@Serializable
data class InterviewQuestion(
    val id: Int,
    val question: String,
    val options: List<String>? = null,
    val required: Boolean = true,
    val answer: String? = null
)

/**
 * Intent understanding data
 */
@Immutable
@Serializable
data class IntentUnderstanding(
    val intent: String,
    val confidence: Float,
    val entities: Map<String, String> = emptyMap(),
    val suggestedActions: List<String> = emptyList()
)

/**
 * Intent confirmation status
 */
enum class ConfirmStatus {
    PENDING,
    CONFIRMED,
    REJECTED,
    MODIFIED
}

/**
 * Thinking stage for AI processing visualization
 */
enum class ThinkingStage {
    ANALYZING,
    PLANNING,
    GENERATING,
    REVIEWING,
    EXECUTING
}

/**
 * Progress information for long-running operations
 */
@Serializable
data class ProgressInfo(
    val step: Int,
    val total: Int,
    val message: String,
    val percentage: Int = if (total > 0) (step * 100 / total) else 0
)

/**
 * Sealed class representing different message types
 * Aligned with PC desktop-app-vue messageTypes.ts
 */
sealed class ChatMessageType {
    /** Standard user message */
    data object User : ChatMessageType()

    /** AI assistant response */
    data object Assistant : ChatMessageType()

    /** System message */
    data object System : ChatMessageType()

    /** Intent recognition result */
    data class IntentRecognition(
        val intent: String,
        val confidence: Float = 1.0f
    ) : ChatMessageType()

    /** Intent confirmation request/response */
    data class IntentConfirmation(
        val understanding: IntentUnderstanding,
        val status: ConfirmStatus = ConfirmStatus.PENDING
    ) : ChatMessageType()

    /** Interview questions for task clarification */
    data class Interview(
        val questions: List<InterviewQuestion>,
        val currentIndex: Int = 0
    ) : ChatMessageType()

    /** Task plan display */
    data class TaskPlanType(
        val plan: TaskPlan,
        val status: PlanningState = PlanningState.IDLE
    ) : ChatMessageType()

    /** Progress update */
    data class Progress(
        val info: ProgressInfo
    ) : ChatMessageType()

    /** Error message */
    data class Error(
        val message: String,
        val code: String? = null,
        val recoverable: Boolean = true
    ) : ChatMessageType()

    /** File reference message */
    data class FileReferenceType(
        val files: List<FileReference>
    ) : ChatMessageType()

    /** Code block message */
    data class CodeBlock(
        val language: String,
        val code: String,
        val filename: String? = null
    ) : ChatMessageType()

    /** Execution result message */
    data class ExecutionResult(
        val success: Boolean,
        val output: String,
        val duration: Long? = null
    ) : ChatMessageType()
}

/**
 * Enhanced chat message with full type support
 */
@Immutable
data class EnhancedChatMessage(
    val id: String = UUID.randomUUID().toString(),
    val role: MessageRole,
    val content: String,
    val timestamp: Long = System.currentTimeMillis(),
    val model: String? = null,
    val tokenUsage: TokenUsage? = null,
    val messageType: ChatMessageType = when (role) {
        MessageRole.USER -> ChatMessageType.User
        MessageRole.ASSISTANT -> ChatMessageType.Assistant
        MessageRole.SYSTEM -> ChatMessageType.System
    },
    val contextMode: ContextMode = ContextMode.PROJECT,
    val referencedFiles: List<FileReference> = emptyList(),
    val taskPlan: TaskPlan? = null,
    val thinkingStage: ThinkingStage? = null,
    val isStreaming: Boolean = false,
    val hasError: Boolean = false,
    val error: String? = null,
    val parentMessageId: String? = null,
    val metadata: Map<String, String> = emptyMap(),
    // Streaming progress
    val streamProgress: Float? = null,
    val streamTokenCount: Int? = null
)

/**
 * Message role enum
 */
enum class MessageRole {
    USER,
    ASSISTANT,
    SYSTEM
}

/**
 * Creation step for AI project creation progress
 */
@Immutable
@Serializable
data class CreationStep(
    val id: Int,
    val name: String,
    val description: String,
    val status: StepStatus = StepStatus.PENDING
)

/**
 * Step status for creation progress
 */
enum class StepStatus {
    PENDING,
    IN_PROGRESS,
    COMPLETED,
    FAILED
}

/**
 * AI creation progress state
 */
@Immutable
data class CreationProgress(
    val steps: List<CreationStep>,
    val currentStep: Int = 0,
    val isComplete: Boolean = false,
    val error: String? = null
)
